### PR TITLE
Add documentation on how to add flags.

### PR DESCRIPTION
**Pull request by kislaykishore**
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: docs/dev_guide.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Each param in GCSFuse should be supported as both as a CLI flag as well as via
+config file; the only exception being *--config-file* which is supported only as
+a CLI flag for obvious reasons. Please follow the steps below in order to make a
+new param available in both the modes:
+
+1.  Declare the new param in
+    [params.yaml](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/params.yaml#L4).
+    Refer to the documentation at the top of the file for guidance.
+1.  Run `make build` to generate the required code.
```
Comment (raj-prince) [inline docs/dev_guide.md:13]: Can we be a bit more explicit, from which directory we need to run the `make build` command?

File: docs/dev_guide.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    and
+    [rationalize_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cfg/rationalize_test.go)
+    and composite tests in
+    [config_validation_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/config_validation_test.go),
+    [config_rationalization_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/config_rationalization_test.go)
+    and
+    [root_test.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/cmd/root_test.go)
+1.  Add the name of flag with underscores in
+    [mount_gcsfuse/main.go](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/mount_gcsfuse/main.go)
+1.  Unless it's a hidden or a deprecated flag, add the flag in Cloud docs for
```
Comment (raj-prince) [inline docs/dev_guide.md:34]: Should we modify this step, as external contributor will not have access to modify the doc?

**Review by raj-prince — COMMENTED** [review]
a few minor comments.

File: docs/dev_guide.md (hunk 2 above)
Comment (kislaykishore) [reply docs/dev_guide.md:34]: Removed it.

**Review by raj-prince — APPROVED** [review]

**Pull request merged**